shared/profile/friendships.native.js

```diff
@@ -46,9 +46,36 @@ const userEntryUsernameStyle = followsYou => ({
   textAlign: 'center',
 })
 
-class FriendshipsRender extends Component<void, Props, void> {
+type State = {
+  dataSource: any,
+}
+
+class FriendshipsRender extends Component<void, Props, State> {
+  state: State ={
+    dataSource: null,
+  }
   _dataSource = new NativeListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2})
 
+  _setDataSource = (props) => {
+    const data = props.currentTab === 'Followers' ? props.followers : props.following
+    const dataSource = this._dataSource.cloneWithRows(_.chunk(data || [], 3))
```

Comment:
Is there a reason you want to use this._dataSource instead of this.state.dataSource?

it also might be a little better to have 2 datasources one for followers and one for following. Probably doesn't matter much, but if we were to hide the tabs instead of unmount them then it would save some thrashing I think